src/AmplifyMapLibreRequest.ts

```diff
@@ -117,13 +117,13 @@ export default class AmplifyMapLibreRequest {
       url = `https://maps.geo.${this.region}.amazonaws.com/maps/v0/maps/${url}/style-descriptor`;
     }
 
-    if (new URL(url).hostname.endsWith(".amazonaws.com")) {
+    const urlObject = new URL(url);
```

Comment:
URL will throw if there isn't a valid URL provided. Do we always know we'll receive a valid url, or do we want this to throw?

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
Also could use a better name than urlObject.

Reply:
By this step in the function, we should have a valid url as we are signing it with AWS keys, so I think we can throw when url is invalid.

Reply:
As I've observed, `transformRequest` is usually called for first time with `url` as map name, like `map5d666777-staging`, which is then used to construct the url for style-description, which is called to get the resource urls for tiles or sprites for the map rendering.`transformRequest` is called again with `url` as something like `https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/map5d666777-staging/sprites/...` to then fetch the tiles or sprites.

At this step in the function, we should have a valid URL to sign with AWS keys so we can throw when URL is invalid.

Reply:
Can we rename the variable on line 117 rather than mutating it. We can have it default to empty string and keep the same logic where it will throw if it's invalid

Reply:
added a new variable to simplify code